src/kirin/dialects/func/attrs.py

```diff
@@ -12,10 +12,10 @@
 
 from ._dialect import dialect
 
-TypeofMethodType = types.PyClass[Method]
-MethodType = types.Generic(
-    Method, types.TypeVar("Params", types.Tuple), types.TypeVar("Ret")
-)
+# TypeofMethodType = types.PyClass[Method]
```

Comment:
Can we get rid of the `Signature` attribute too? Now you have a function type we can just use that to replace the `Signature` attribute